gammapy/maps/core.py

```diff
@@ -981,6 +981,61 @@ def _resample_by_idx(self, idx, weights=None, preserve_counts=False):
         """
         pass
 
+    def reproject_spatial_geom(self, geom, preserve_counts=False, oversampling_factor=10):
+        """Reproject map to input geometry using oversampling :
+           the base map is oversampled and its bins are re-distributed
+           into a map with the target geom.
+           Note that alternative methods can be found in the reproject package.
+        
+        Parameters
+        ----------
+        geom : `~gammapy.maps.Geom`
+            2D Map geometry
+        preserve_counts : bool
+            Preserve the integral over each bin.  This should be true
+            if the map is an integral quantity (e.g. counts) and false if
+            the map is a differential quantity (e.g. intensity)
+        oversampling_factor : int
+           Minimal factor between the bin size of the output map and the oversampled base map.
+           Used only for the oversampling method.
+
+        Returns
+        -------
+        output_map : `Map`
+            Reprojected Map
+        """
+        from .region import RegionGeom
+        from .hpx import HpxGeom
+
+        if isinstance(geom, RegionGeom) or isinstance(self.geom, RegionGeom):
+            raise TypeError(f"Reproject from {type(self.geom)} to {type(geom)} is not supported")
+
+        
+        if geom.is_image:
+            axes = [ax.copy() for ax in self.geom.axes]
+            geom = geom.copy(axes=axes)
+        else:
+            raise TypeError("Invalid target geometry, use geom.to_image()")
+
+        output_map = Map.from_geom(geom, unit=self.unit)
+        base_factor = geom.pixel_scales.min()/self.geom.pixel_scales.min()
```

Comment:
Maybe simplify here and just always apply the `oversampling_factor` to `self`? One could e.g. choose a factor of 2 as default as a compromise...

Reply:
I think the comment is obsolete after we agreed to keep the "adaptive resampling factor" and rename it from `oversampling_factor` to something else.